flink-scala-shell/src/main/java/org/apache/flink/api/java/ScalaShellRemoteEnvironment.java

```diff
@@ -19,27 +19,42 @@
  * limitations under the License.
  */
 
+import org.apache.flink.api.common.InvalidProgramException;
 import org.apache.flink.api.common.JobExecutionResult;
 import org.apache.flink.api.common.Plan;
 import org.apache.flink.api.common.PlanExecutor;
 import org.apache.flink.api.scala.FlinkILoop;
 import org.apache.flink.configuration.Configuration;
 
+import java.io.File;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /**
- * Special version of {@link org.apache.flink.api.java.RemoteEnvironment} that has a reference
+ * Special version of {@link org.apache.flink.api.java.ExecutionEnvironment} that has a reference
  * to a {@link org.apache.flink.api.scala.FlinkILoop}. When execute is called this will
  * use the reference of the ILoop to write the compiled classes of the current session to
  * a Jar file and submit these with the program.
  */
-public class ScalaShellRemoteEnvironment extends RemoteEnvironment {
+public class ScalaShellRemoteEnvironment extends ExecutionEnvironment {
```

Comment:
Aha, this is what I want to do in [FLINK-14629](https://issues.apache.org/jira/browse/FLINK-14629). Thanks for picking it up.